plugins/src/hashlookup/hashlookup.go

```diff
@@ -0,0 +1,492 @@
+/*
+ * Template to develop new plugins.
+ * Check GUI documentation section "Administration" for a step-by-step workflow
+ */
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/Jeffail/gabs/v2"
+	"github.com/blastrain/vitess-sqlparser/sqlparser"
+	"github.com/cert-lv/graphoscope/pdk"
+	"github.com/umpc/go-sortedmap"
+	"github.com/umpc/go-sortedmap/desc"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+	"sync"
+)
+
+/*
+ * Check "pdk/plugin.go" for the built-in plugin functions description
+ */
+
+func (p *plugin) Source() *pdk.Source {
+	return p.source
+}
+
+func (p *plugin) Setup(source *pdk.Source, limit int) error {
+
+	// Validate necessary parameters
+	if source.Access["url"] == "" {
+		return fmt.Errorf("'access.url' is not defined")
+	} else if source.Access["url"][0:4] != "http" {
+		return fmt.Errorf("'access.url' must start with 'http[s]://'")
+	}
+
+	// Store settings
+	p.source = source
+	p.limit = limit
+	p.url = source.Access["url"]
+
+	// Set possible variable type & searching fields
+	for _, relation := range source.Relations {
+		for _, types := range relation.From.VarTypes {
+			types.RegexCompiled = regexp.MustCompile(types.Regex)
+		}
+
+		for _, types := range relation.To.VarTypes {
+			types.RegexCompiled = regexp.MustCompile(types.Regex)
+		}
+	}
+
+	fmt.Printf("HTTP %s: %#v\n\n", source.Name, p)
+	return nil
+}
+
+func (p *plugin) Search(stmt *sqlparser.Select) ([]map[string]interface{}, map[string]interface{}, error) {
+
+	// Storage for the results to return
+	results := []map[string]interface{}{}
+
+	searchFields, err := p.convert(stmt)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var bodies []*bytes.Buffer
```

Comment:
`debug` info was recently added as returned value. Will also be visible in browser console if enabled
```go
searchFields, err := p.convert(stmt)
if err != nil {
	return nil, nil, nil, err
}

// Debug info
debug := make(map[string]interface{})
var bodies []*bytes.Buffer
```